core/src/main/java/io/questdb/cairo/TableUtils.java

```diff
@@ -1133,6 +1133,24 @@ public static int openFileRWOrFail(FilesFacade ff, LPSZ path, long opts) {
         return openRW(ff, path, LOG, opts);
     }
 
+    /**
+     * Opens and loads a metadata file.
+     *
+     * @param configuration
+     * @param tableToken
+     * @return
+     */
+    public static TableReaderMetadata openMetadataReader(@NotNull CairoConfiguration configuration, @NotNull TableToken tableToken) {
```

Comment:
nit: this code refactoring may not be necessary and can be reverted.

Reply:
It's better to move it back to `TableReader`, this method is used in 1 class, there is no `utility` in this `util`. Also it adds unnecessary coupling between `TableUtils` and `TableReaderMetadata`